scripts/bfb-install

```diff
@@ -27,130 +27,541 @@
+  fi
+}
+
+# Run a command locally or remotely via SSH
+#
+# $1: mode (local or remote)
+# $2: command
+#
+# Global variables used:
+#   $ip, $sudo_prefix, $run_cmd_local_ready, $run_cmd_remote_ready
+#
+# Example:
+#   run_cmd local "ls -l"
+#   run_cmd remote "ls -l"
+run_cmd()
+{
+  if [ $# -lt 2 ]; then
+    echo "Error: run_cmd() needs at least 2 arguments."
+    exit 1
+  fi
+
+  local mode=$1
+  local command=$2
+
+  echo_dbg "Running command in $mode mode: $command"
+
+  if [ "$mode" == "local" ]; then
+    if [ $run_cmd_local_ready -eq 0 ]; then
+      echo "Error: 'run_cmd local' are not ready to use"
+      exit 1
+    fi
+    $sudo_prefix sh -c "$command"
+  elif [ "$mode" == "remote" ]; then
+    if [ $run_cmd_remote_ready -eq 0 ]; then
+      echo "Error: 'run_cmd remote' are not ready to use"
+      exit 1
+    fi
+    # Execute the command networkly via SSH
+    ssh root@$ip "$command"
+  else
+    echo "Error: invalid mode: $mode"
+    return 1
+  fi
+}
+
+# Run a command locally or remotely via SSH and exit on error with a custom message
+# $1: mode (local or remote)
+# $2: command
+# $3: custom error message (optional)
+#
+# Global variables used: $ip
+run_cmd_exit()
+{
+  if [ $# -lt 2 ]; then
+    echo "Error: run_cmd_exit() requires at least 2 arguments."
+    exit 1
+  fi
+
+  local mode=$1
+  local command=$2
+  local error_msg="${3:-"Error: Command failed"}: [$command]"
+
+  run_cmd "$mode" "$command"
+  local RETVAL=$?
+
+  if [ $RETVAL -ne 0 ]; then
+    echo "$error_msg"
+    exit $RETVAL
+  fi
+}
+
+check_local_netcat_exit()
+{
+  check_netcat_exit "local"
+}
+
+check_remote_netcat_exit()
+{
+  check_netcat_exit "remote"
+}
+
+# Push the boot stream to rshim via local rshim
+#
+# Global variables used:
+#   $bfb, $cfg, $rootfs, $pv, $rshim_node, $sudo_prefix   
+push_boot_stream_via_local_rshim()
+{
+  # Push the boot stream to local rshim
+  echo "Pushing bfb${cfg:+ + cfg}${rootfs:+ + rootfs}"
+  $sudo_prefix sh -c "cat ${bfb} ${cfg:+$cfg} ${rootfs:+${rootfs}} ${pv:+| ${pv} | cat -} > ${rshim_node}/boot"
+}
+
+# Push the boot stream to rshim via remote rshim with scp
+#
+# Global variables used
+#   $bfb, $cfg, $rootfs, $pv, $ip, $rshim
+push_boot_stream_via_remote_rshim_scp()
+{
+  # Push the boot stream to remote rshim via ssh copy
+  echo "Pushing bfb${cfg:+ + cfg}${rootfs:+ + rootfs} to ${ip} via scp"
+  sh -c "cat ${bfb} ${cfg:+$cfg} ${rootfs:+${rootfs}} ${pv:+| ${pv} | cat -} | ssh root@$ip \"cat > ${rshim_node}/boot\""
+}
+
+# Push the boot stream to rshim via remote rshim with netcat
+#
+# Global variables used:
+#   $bfb, $cfg, $rootfs, $pv, $ip, $port, $rshim_node
+push_boot_stream_via_remote_rshim_nc()
+{
+  data="${bfb} ${cfg:+$cfg} ${rootfs:+${rootfs}}"
+  # Push the boot stream to remote rshim via netcat
+  echo "Pushing $data to ${ip} via netcat"
+
+  echo "Starting a remote netcat server..."
+  start_nc_server_cmd="nc -l -p $port > ${rshim_node}/boot &"
+  run_cmd_exit remote "$start_nc_server_cmd"
```

Comment:
Have you tried this remote rshim from one server to another server (instead of rshim)? line 179-180 run ssh command and exit. Just curious whether the "nc" command will terminate if the parent process (ssh) exit on a standard server. If this is the case, you might consider adding 'nohup' before the 'nc'.

Reply:
The new PR commit (67d8ab7) has been tested between Server and BMC as well between Server and Server. There were indeed some issues initially with the SSH execution not terminating as well as other issues which adding "nohup" is part of the solution.